### ISSUE TITLE
Cop idea: Redundant safe navigation for basic types

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12412]
## Describe the solution you'd like

1. 
```ruby
hsh = [['a', '1'], ['b', '2']]
# hsh = nil

# bad
hsh&.to_h { |k, v| [k, v] } || {}
# bad
hsh&.to_h || {}

# good
hsh.to_h { |k,v| [k, v] }
# good
hsh.to_h
```

2.
```ruby
arr = ['a', 'b', 'c']
# arr = nil

# bad
arr&.to_a || []

# good
arr.to_a
```

3.
```ruby
str = "foo"
# str = nil

# bad
str&.to_s || ''

# good
str.to_s
```

**Comment by koic** [comment]
Rather than creating a new cop, extending the existing `Lint/RedundantSafeNavigation` should perhaps be considered."

**Comment by fatkodima** [comment]
I do not think we need a new cop, `Lint/RedundantSafeNavigation` can be extended, as it was once extended for constants.
